### PR TITLE
mm_network: Add debug and failsafes for DNS config

### PR DESCRIPTION
**Pull request by okurz**
Verification run: https://openqa.opensuse.org/tests/995938#step/before_test/45

Related progress issue: https://progress.opensuse.org/issues/52499

### CODE REVIEW

**Comment by ggardet** [comment]
IIRC, ping is blocked on default qemu network, by default. Not sure about tap config.

File: lib/mm_network.pm [hunk 1]
```diff
@@ -79,10 +79,12 @@ sub configure_default_gateway {
 
 sub configure_static_dns {
     my ($conf) = @_;
-    my $servers = join(" ", @{$conf->{nameserver}});
-    script_run("sed -i -e 's|^NETCONFIG_DNS_STATIC_SERVERS=.*|NETCONFIG_DNS_STATIC_SERVERS=\"$servers\"|' /etc/sysconfig/network/config");
-    script_run("netconfig -f update");
-    script_run("cat /etc/resolv.conf");
+    my @servers = $conf->{nameserver};
+    assert_script_run("ping -c 1 $_") foreach @servers;
```
Comment (ggardet) [inline lib/mm_network.pm]: @okurz it tries to ping an array: https://openqa.opensuse.org/tests/995903#step/before_test/45

**Comment by okurz** [comment]
@asmorodskyi ?

**Comment by asmorodskyi** [comment]
1. agree to replace script_run with assert_script_run 
2. disagree with ping from several perspectives : 
 - this goes against idea why we actually providing several DNS servers - to increase chances for hostname to be resolved if **at least one** DNS is alive 
 - currently in wicked tests we going in opposite direction because we facing a lot situations when link is actually up and if you send 100 packets you will receive 99-95 of them but such kind of ping when you sending single packet is in bad luck to be lost. Of course generally it is unacceptable to loose packets without good reason but keep in mind that this function might be used by some tests which should fail when there is network hiccups so for **this** function I would say that  it is OK to lost few packets to **some** of DNS servers
 
if you really want to check that setup DNS function did a job - try to resolve some FQDN instead of pinging servers

**Comment by okurz** [comment]
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8162

**Pull request closed without merging**